x-pack/plugins/enterprise_search/public/applications/app_search/app_logic.ts

```diff
@@ -26,5 +33,93 @@ export const AppLogic = kea<MakeLogicType<IAppValues, IAppActions>>({
         initializeAppData: () => true,
       },
     ],
+    account: [
+      {} as IAccount,
+      {
+        initializeAppData: (_, { appSearch: account }) => account,
+        setOnboardingComplete: (state) => ({
+          ...state,
+          onboardingComplete: true,
+        }),
+      },
+    ],
+    configuredLimits: [
+      {} as IConfiguredLimits,
```

Comment:
I think I'm losing it. I swear I left a comment here but now it's gone, lost in the Github ether. Oh well, typing again, sorry if you get a notification about this twice.

My question is, how is it valid to do `{} as IConfiguredLimits` 🤔 . The IConfiguredLimits interface requires an engines property. You are using an empty object that has no engines property. What happens if a consumer of this logic is expecting an IConfiguredLimits which according to the Interface ALWAYS has an engines property, and it's not there?

I am probably missing something. Can you shed some light on this?

Reply:
Super good q - worst case scenario we'd use `optional?.chaining?.syntax` wherever possible, but I'm also fine with changing this default to something like `{ engines: {} }` if you prefer!

Also for context, the engines obj is a pretty recent change in our API syntax (courtesy of Bmac, now that the API sends other non-engine related limits down the wire), so it definitely makes sense to update to accommodate that.

Reply:
I don't whether it's a problem or not... I think I'm more just surprised that TS allows it. 

What about defaulting to `null`?

Reply:
> I think I'm more just surprised that TS allows it

It's because `as IConfiguredLimits` coerces Typescript to "read" it as valid. Contrived example:

```
const someObj: IConfiguredLimits = {}; // will trigger a typescript error
```

```
const someObj = {} as IConfiguredLimits; // will not trigger a typescript error
```

Hope that makes sense!

Reply:
I get it, but I don't think it's a good idea. I left a comment about that in the other thread.

Reply:
> What about defaulting to `null`?

Looping @byronhulcher in on this one - IIRC, we _used_ to have null defaults for a lot of our logic files, but we ended up switching to empty objs for obj values. While it's less convenient for quick `if (!someObj)` checks, it's also more convenient for diving children, e.g. `if (someObj.someChild)` vs `if (someObj && someObj.someChild)` which is in some of our older code.

At this point, I think `{}` defaults is the convention for most of our most recent logic files (Byron prob knows better than me), so we'd have to change it across the board. 🤔